src/rqt_gauges_2/steering_wheel_gauge.py

```diff
@@ -0,0 +1,160 @@
+import math
+
+from PyQt5.QtWidgets import QWidget
+from PyQt5.QtGui import QPolygon, QPolygonF, QColor, QPen, QFont, QPainter, QFontMetrics, QConicalGradient, QRadialGradient, QBrush
+from PyQt5.QtCore import Qt, QPoint, QPointF, QObject, QRect
+
+class SteeringWheelGauge(QWidget):
+    # This class creates a gauge object that contains several elements such as a needle, big and small scales
+    # and different circles painted with different colors to look like a car Speedometer. It has the ability to
+    # modify the minimum and maximum values of the gauge and the units the numbers are displaying. The class contains
+    # methods used to modify the values explained before, the marked number of the gauge and the whole design as well.
+
+    def __init__(self, parent=None):
+        # Constructor method of the class, initializes all the variables needed to create the gauge.
+
+        super(SteeringWheelGauge, self).__init__(parent)
+
+        self.value = 0
+        self.width = 400
+        self.height = 400
+        self.progress_width = 25
+        self.progress_rounded_cap = True
+        self.progress_color = 0x39F030
+        self.max_value = 45
+        self.circle_max_angle = 135
+        self.font_size = 40
+        self.scale_font_size = 15
+        self.font_family = "Verdana"
+        self.suffix = "°"
+        self.text_color = 0x000000
+        self.enable_shadow = True
+
+        self.angle_offset = 0
+
+        self.minValue = -45
+        self.maxValue = 45
+
+        if self.width <= self.height:
+            self.widget_diameter = self.width
+        else:
+            self.widget_diameter = self.height
+
+        self.resize(self.width, self.height)
+
+
+    def draw_big_scaled_marker(self, start_angle_value, angle_size, scalaCount):
+        painter = QPainter(self)
+        painter.setRenderHint(QPainter.Antialiasing)
+        painter.translate(self.width / 2, self.height / 2)
+
+        self.pen = QPen(Qt.black)
+        self.pen.setWidth(2)
+        painter.setPen(self.pen)
+
+        painter.rotate(start_angle_value - self.angle_offset)
+        steps_size = (float(angle_size) / float(scalaCount))
+        scale_line_outer_start = (self.widget_diameter)/2
+        scale_line_lenght = (self.widget_diameter / 2) - (self.widget_diameter / 20)
```

Comment:
Can we pre-compute these values in the init?

Reply:
We can. Done.